### ISSUE TITLE
Errors on startup of my application with version v.0.6 (dev)

### ISSUE THREAD

**Issue by Bram94** [vispy/vispy#1396]
After having upgraded to the dev v.0.6 version, I always get the following error (error 1) during the startup of my application, causing it to crash. It seems related to issue #1277, where I also initially posted these errors, but as it is not the only type of error I sometimes receive (see post below), I have opened this new issue for it.

I checked whether the problem would be solved when removing the lines in which GL_LINE_SMOOTH is enabled/disabled, in the function _prepare_draw in visuals/line/line.py. This of course removed the error related to GL_LINE_SMOOTH, but then I got another error related to setting the line width (error 2).
After removing also those lines, just to check whether that would work, I got however another error (error 3).
I don't know whether this is al related to the same bug (with GL_LINE_SMOOTH), or if it involves multiple bugs. I put it here because the first error is the same as the one in this issue, but if it involves multiple bugs, then it could be moved.
Does anyone have ideas about how to solve this, or maybe a good way to circumvent this problem?

1:
```
File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 455, in __call__
    self._invoke_callback(cb, event)
File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 475, in _invoke_callback self, 
    cb_event=(cb, event)) << caught exception here: >>
File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 471, in _invoke_callback 
    cb(event)
File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 208, in on_draw
    self._draw_scene()
File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 254, in _draw_scene
    self.draw_visual(self.scene)
File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 292, in draw_visual 
    node.draw()
File "D:\Python\NLradar_eigen\Python_files\vispy\scene\visuals.py", line 98, in draw
    self._visual_superclass.draw(self)
File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\visual.py", line 588, in draw
    v.draw()
File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\visual.py", line 432, in draw
    if self._prepare_draw(view=self) is False:
File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\line\line.py", line 356, in _prepare_draw
    GL.glDisable(GL.GL_LINE_SMOOTH)
File "C:\Anaconda3\lib\site-packages\OpenGL\platform\baseplatform.py", line 402, in __call__
    return self( *args, **named )
File "C:\Anaconda3\lib\site-packages\OpenGL\error.py", line 232, in glCheckError
    baseOperation = baseOperation, OpenGL.error.GLError: GLError( err = 1282, description = b'ongeldige bewerking', baseOperation = glDisable, cArguments = (GL_LINE_SMOOTH,)
```

2:
```
OpenGL.error.GLError: GLError( err = 1282, description = b'ongeldige bewerking', baseOperation = glLineWidth, cArguments = (1.0,)
```

3:
```
D:\Python\NLradar_eigen\Python_files\vispy\visuals\visual.py:442: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead logger.warn("Error drawing visual %r" % self) WARNING: Error drawing visual <vispy.visuals.line.line._GLLineVisual object at 0x000001E803483828> WARNING: Traceback (most recent call last):
File "nlr.py", line 2367, in <module>
    main()
File "nlr.py", line 2364, in main
    sys.exit(app.exec_())
File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 435, in event out =
    super(QtBaseCanvasBackend, self).event(ev)
File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 707, in paintGL
    self._vispy_canvas.events.draw(region=None)
File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 455, in __call__
    self._invoke_callback(cb, event) File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 475, in _invoke_callback
    self, cb_event=(cb, event)) << caught exception here: >>
File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\gl\_gl2.py", line 735, in glGetShaderInfoLog
    nativefunc = glGetShaderInfoLog._native AttributeError: 'function' object has no attribute '_native'
```

**Comment by Bram94** [comment]
An addition:
I get these errors not only with the new v.0.6 version, but also infrequently with the previous dev v.0.5 version that I used. When displaying 8 panels simultaneously, then it takes on average about 2-5 minutes of constant use of my application until it crashes because of an error. With v.0.6 my application always crashes on startup, during the plotting of line visuals.
Sometimes this is the error related to GL_LINE_SMOOTH, which is shown above, but sometimes it is one the 2 errors shown below.

Because with the previous v.0.5 version, the call of glDisable(GL_LINE_SMOOTH) usually does not lead to errors, it might in fact be the case that it is not the main call that causes the errors. I have also noticed that the occurrence of this error is always accompanied by another error when setting glLinewidth (error 2 in my post above), further strengthening the idea that the problem is not just the call of glDisable(GL_LINE_SMOOTH).

It is a very frustrating bug, among others because it is difficult to reproduce due to the infrequent occurrence. I also don't have any idea about where to look for causes unfortunately, due to my lack of knowledge of OpenGL, and my limited knowledge of vispy. I will now continue using the old dev v.0.5 version, and will try to properly catch the errors mentioned above, to prevent the crashing of my application upon encountering them.

`WARNING: Traceback (most recent call last): File "nlr.py", line 2716, in <module> main() File "nlr.py", line 2713, in main sys.exit(app.exec_()) File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 433, in event out = super(QtBaseCanvasBackend, self).event(ev) File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 433, in event out = super(QtBaseCanvasBackend, self).event(ev) File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 705, in paintGL self._vispy_canvas.events.draw(region=None) File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 455, in __call__ self._invoke_callback(cb, event) File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 476, in _invoke_callback self, cb_event=(cb, event)) << caught exception here: >> File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 471, in _invoke_callback cb(event) File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 214, in on_draw self._draw_scene() File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 259, in _draw_scene self.draw_visual(self.scene) File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 293, in draw_visual node.draw() File "D:\Python\NLradar_eigen\Python_files\vispy\scene\visuals.py", line 98, in draw self._visual_superclass.draw(self) File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\visual.py", line 433, in draw if self._prepare_draw(view=self) is False: File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\markers.py", line 621, in _prepare_draw view.view_program['u_viewport']=gl.glGetParameter(gl.GL_VIEWPORT) File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\program.py", line 317, in __setitem__ 'not %i.' % (name, numel, data.size)) ValueError: Uniform 'u_viewport' needs 4 elements, not 0.`

`WARNING: Error drawing visual <Text at 0x17c118739b0> WARNING: Traceback (most recent call last): File "nlr.py", line 2716, in <module> main() File "nlr.py", line 2713, in main sys.exit(app.exec_()) File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 433, in event out = super(QtBaseCanvasBackend, self).event(ev) File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 433, in event out = super(QtBaseCanvasBackend, self).event(ev) File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 705, in paintGL self._vispy_canvas.events.draw(region=None) File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 455, in __call__ self._invoke_callback(cb, event) File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 476, in _invoke_callback self, cb_event=(cb, event)) << caught exception here: >> File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\gl\_gl2.py", line 403, in glDisableVertexAttribArray nativefunc = glDisableVertexAttribArray._native AttributeError: 'function' object has no attribute '_native'`

**Comment by davidh-ssec** [comment]
@Bram94 Thanks for the bug report. FYI I updated the formatting in your post to make it easier to read (use three backticks before and after a literal block).

First thing that would help is if you could provide the output of `print(vispy.sys_info()` so we can figure out what version of OpenGL you have supported.

Now for the few guesses I have about what's causing some of these. First, the `'function' object has no attribute '_native'` is most certainly happening because of previous errors happening when creating the canvas/app or other portions of the vispy system.

Second, on 0.6.0dev (the master branch of this github repository) I have implemented some changes/fixes to the Ellipse and Rectangle visuals. Most importantly I fixed the fact that the "border_width" parameter was not being passed to the parent PolygonVisual object. Meaning the border width had no effect in a RectangleVisual. This would be a good guess as to why you are *always* getting the error after removing the line smoothing.

How are you creating vispy elements (gloo, Scene and Visuals, or the Fig/plotting API)? If with Visuals I'm guessing you are using the Ellipse, Rectangle, or Line visuals? A quick hack that you could try would be to try `my_vis._border.method = 'agg'` for the Ellipse and Rectangles and `my_line.method = 'agg'` for the Line visuals.

**Comment by kmuehlbauer** [comment]
@davidh-ssec This might not have anything to do with the errors shown, but I wonder why profiling is activated in the LineVisual.

**Comment by davidh-ssec** [comment]
Yeah I brought that up on gitter a long while ago and didn't get a response. I'm also researching line drawing techniques with triangles and other things (not sure if that is different than the current 'agg'). We may have to raise an error or warn and provide a change in default for the line drawing method based on `antialias` and `width` parameters provided by the user. By that I mean if you have 1px width lines then use the default OpenGL method, anything else use 'agg' or other new method.

**Comment by Bram94** [comment]
@davidh-ssec Thanks for setting the priority to 'high', and for the reformatting of my posts. I have included the month from which my dev v.0.5 version dated in my second post, which is december 2016. 

Here is the output of vispy.sys_info() (I assume that you don't need the rest of the output):
``` 
Platform: Windows-10-10.0.15063-SP0
Python:   3.5.4 |Anaconda custom (64-bit)| (default, Sep 27 2017, 18:54:39) [MSC v.1900 64 bit (AMD64)]
NumPy:    1.11.3
Backend:  PyQt5
pyqt4:    None
pyqt5:    ('PyQt5', '5.6', '5.6.2')
pyside:   None
pyglet:   None
glfw:     None
sdl2:     None
wx:       None
egl:      None
osmesa:   None
_test:    None

GL version:  '4.5.0 NVIDIA 369.09'
MAX_TEXTURE_SIZE: 16384
Extensions: 'GL_AMD_multi_draw_indirect
```

About the method for creation of visuals, and the visuals that I use: Of the 3 types of visuals that you mention I use only the Line visuals. I have tried to change the method from 'gl' to 'agg', but besides that it is problematic due to the fact that I need to use a numpy array for the 'connect' attribute, I also get other errors when doing that (with connect='strip'), which are however related to the Text visual. These are the one below and error 2 in my second post above, about the '_native' attribute.

``` 
WARNING: Error drawing visual <Text at 0x252e4d471d0>
WARNING: Traceback (most recent call last):
  File "nlr.py", line 2709, in <module>
    main()
  File "nlr.py", line 2706, in main
    sys.exit(app.exec_())
  File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 435, in event
    out = super(QtBaseCanvasBackend, self).event(ev)
  File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 435, in event
    out = super(QtBaseCanvasBackend, self).event(ev)
  File "D:\Python\NLradar_eigen\Python_files\vispy\app\backends\_qt.py", line 707, in paintGL
    self._vispy_canvas.events.draw(region=None)
  File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 455, in __call__
    self._invoke_callback(cb, event)
  File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 475, in _invoke_callback
    self, cb_event=(cb, event))
  << caught exception here: >>
  File "D:\Python\NLradar_eigen\Python_files\vispy\util\event.py", line 471, in _invoke_callback
    cb(event)
  File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 208, in on_draw
    self._draw_scene()
  File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 254, in _draw_scene
    self.draw_visual(self.scene)
  File "D:\Python\NLradar_eigen\Python_files\vispy\scene\canvas.py", line 292, in draw_visual
    node.draw()
  File "D:\Python\NLradar_eigen\Python_files\vispy\scene\visuals.py", line 98, in draw
    self._visual_superclass.draw(self)
  File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\visual.py", line 440, in draw
    self._vshare.index_buffer)
  File "D:\Python\NLradar_eigen\Python_files\vispy\visuals\shaders\program.py", line 77, in draw
    Program.draw(self, *args, **kwargs)
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\program.py", line 470, in draw
    canvas.context.flush_commands()
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\context.py", line 171, in flush_commands
    self.glir.flush(self.shared.parser)
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\glir.py", line 232, in flush
    self._shared.flush(parser)
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\glir.py", line 146, in flush
    parser.parse(self._filter(self.clear(), parser))
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\glir.py", line 477, in parse
    self._parse(command)
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\glir.py", line 447, in _parse
    ob.set_data(*args)
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\glir.py", line 976, in set_data
    gl.check_error('periodic check')
  File "D:\Python\NLradar_eigen\Python_files\vispy\gloo\gl\__init__.py", line 211, in check_error
    raise err
RuntimeError: OpenGL got errors (periodic check): GL_INVALID_OPERATION
```

**Comment by kmuehlbauer** [comment]
@Bram94 Is it possible for you to use PyQt4?

**Comment by Bram94** [comment]
@kmuehlbauer I will try that, and mention the results here.

**Comment by kmuehlbauer** [comment]
@Bram94 Does the example from #1277 also break for you? It works for me under Linux with PyQt4.

**Comment by Bram94** [comment]
I have in the mean time tested using PyQt4, and it seems that everything works properly under it. I haven't encountered any error related to vispy despite a decent amount of testing, and I also don't get the GL_LINE_SMOOTH error upon startup with the dev v.0.6 version. 
It must therefore be the case that that error is related to the use of PyQt5, and the other errors that I got might also have been related to it, because I haven't received them yet under PyQt4. 

@kmuehlbauer I have tested the example from #1277, and it doesn't give an error for me, but the app also fails to startup (I get the message that it does not respond).

**Comment by kmuehlbauer** [comment]
@Bram94 Thanks for checking. On my linux box the example works on both versions PyQt4 and PyQt5. So this might be kind of a non-linux problem. The only difference I see, is different thickness of the Axis-Line, which is 1px with PyQt5-backend and maybe 3 on the PyQt4.

You didn't get anything on the screen, because the `app` wasn't run:
```
from vispy import scene, app
import sys


canvas = scene.SceneCanvas(keys=None, size=(800, 600), show=True)
view = canvas.central_widget.add_view()
scene_transform = scene.STTransform()
view.camera = scene.cameras.TurntableCamera(parent=view.scene,
                                            fov=0., distance=4.0)

ax = scene.visuals.Axis(pos=[[-1.0, 0], [1.0, 0]],
                        tick_direction=(0, -1),
                        parent=view.scene, axis_label='X',
                        anchors=['center', 'middle'])

canvas.render()

if __name__ == '__main__' and sys.flags.interactive == 0:
    app.run()
```

**Comment by kmuehlbauer** [comment]
> BTW, did something change in the function draw_visual in scene/canvas.py? I call it in my code sometimes to update and determine the dimensions of a widget, but with the new version the size of the widget is not updated anymore after calling draw_visual (at least as obtained by calling widget.size).

The latest changes on that function have been introduced 3 years ago, but after release of 0.4.0. If you were on 0.4.0 before this might be a reason for the behaviour.

If you change the widgets dimension, it normally should be updated. Obviously this is done somewhere else.

**Comment by Bram94** [comment]
@kmuehlbauer The from @astrofrog works for me also with PyQt5, so the issue related to that might be different from my issue. 
Further, I have upgraded PyQt5 to the latest version (v5.9.1), and the problem is still present. 

It is however a good idea to check whether the problem is also apparent on Linux with PyQt5. I can't do that by myself soon, so I will send my code to you by mail @kmuehlbauer, hoping that you are able to test that.

**Comment by kmuehlbauer** [comment]
@Bram94 Good! Can you confirm the difference in axis line thickness between qt4 and qt5 as I see on Linux?

**Comment by Bram94** [comment]
@kmuehlbauer On Windows I don't see such a difference in axis line thickness between qt4 and qt5. Further, I have sent you the code by e-mail.

About my question about draw_visual: This 'problem' appeared to be related to one of my own small changes in the v.0.5 version of vispy, that I did not yet include in the new version.

Currently everything works fine under the v.0.6 version with pyqt4, except that mouse moves (both panning and just moving) are handled very slowly when the amount of panels becomes greater than 4. This seems very strange to me, and I don't think that it's related to one of my own small changes in the vispy code. For scrolling by using the mouse wheel the speed is unchanged. 
@davidh-ssec @kmuehlbauer Are you aware of any change that might cause this behaviour? If not, then I will look again at whether it might be related to a change by myself.

**Comment by Bram94** [comment]
I have reinstalled my Anaconda distribution (because Spyder had failed after some (un)installations of packages), and after doing that I don't get the error on startup anymore with PyQt5. Comparable errors do however still occur after using the application for some time, usually after around 1 minute.
Besides getting these 'invalid operation' errors for the Text and Line visuals (as shown above), I also get them for the Mesh visual, which I didn't mention here yet.

About my remark about slow panning in the case of more than 4 panels with the v.0.6 version: With PyQt5 (which I am now able to test) this appears to be less of an issue, although panning is still clearly slower than with the old v.0.5 version. Under PyQt4 the panning is really annoying slow. 
@davidh-ssec @kmuehlbauer I am still curious about whether you can relate this to some of the changes that have been made to Vispy after december 2016.

**Comment by davidh-ssec** [comment]
@Bram94 Just to be clear about the slow panning: You are saying that with the old v0.5.0dev version you had from December 2016 to the current github master (0.6.0dev) the panning now seems much slower regardless of PyQt5 or PyQt4 backends? For future discussion let's make sure to use "dev" to describe the version from the github master branch whether it was 0.5.0dev Dec2016 or 0.6.0dev now.

After a quick glance I don't see anything that seems like it should have affected camera or even Scene performance, but being able to narrow down the dates/commits of the changes would be very helpful.

**Comment by larsoner** [comment]
[Git bisect](http://webchick.net/node/99) is your friend

**Comment by kmuehlbauer** [comment]
You could also try github's `blame` feature if you can narrow down to some single file.

**Comment by Bram94** [comment]
@davidh-ssec  With 0.6.0dev panning is slower than with 0.5.0dev.
When I have time then I will try to find the bottleneck regarding speed (which will take some time, as I need to learn more about the structure of Vispy).

**Comment by larsoner** [comment]
FWIW the `git bisect` tests you can do without diving into the `vispy` internals. It should show the commit where things slowed down, which implies which code changes caused it.

**Comment by Bram94** [comment]
Then I will try that tonight.

**Comment by kmuehlbauer** [comment]
@larsoner You are completely right, `git bisect` is the tool to use in this case. Just wanted to mention the possible usefulness of github blame in corner cases.

**Comment by Bram94** [comment]
I have found the problem, and it was that I forgot to add a change in 0.6.0dev that I had included in 0.5.0 (which greatly increased the plotting speed for PyQt4 with more than 4 panels).
There is however still a panning speed difference between 0.6.0dev and 0.5.0dev, also after including this change. v.0.6.0dev is much faster than before the change, but still significantly slower than v.0.5.0dev. With use of the git-bisect method mentioned by @larsoner I have been able to pinpoint the decrease in panning speed to the 4 commits between 8 and 11 september by himself. Maybe it's therefore worth taking a look at whether the performed changes can be somewhat optimized?

Because the changes that I made to improve the (mostly panning and zooming) speed might be interesting for others (maybe some of them can also be incorporated in the current Vispy version?), I put them here:
- The 'transforms_changed' event in visuals/transforms/base_transform.py is not emitted during a panning/zooming event. Emitting it caused the very slow panning with more than 4 panels under PyQt4.
- I determine the picked visual in the function _process_mouse_event in scene/canvas.py not by calling the function visual_at (which is slow), but by determining in another way which panel is selected by the mouse, and then 
taking the picked visual to be the viewbox corresponding to the selected panel.
- Not related to speed, but also a useful small change I think: Change in the function zoom in scene/cameras/panzoom.py the line 'assert len(center) in (2, 3, 4)' into 'assert center is None or len(center) in (2, 3, 4)'. center is None implies that the current center of the viewbox is used as the zooming center, which is what I use for example, but the line 'assert len(center) in (2, 3, 4)' prohibits that.

I have no new information yet about the bug for which I started the issue.

**Comment by larsoner** [comment]
>  I have been able to pinpoint the decrease in panning speed to the 4 commits between 8 and 11 september by himself.

Are you unable to run your test code on each of these commits separately? Why not isolate to one (and link to it from [this list](https://github.com/vispy/vispy/commits/master?after=7e7c8c3df8ad4c5a881f1cfef9f292df64d08707+104)?

> The 'transforms_changed' event in visuals/transforms/base_transform.py is not emitted during a panning/zooming event. Emitting it caused the very slow panning with more than 4 panels under PyQt4.

Can you see if this is related to or fixed by #1390? It sounds perhaps related, but if it's not, a separate PR would be good (assuming it works and doesn't break anything of course!).

> I determine the picked visual in the function _process_mouse_event in scene/canvas.py not by calling the function visual_at (which is slow), but by determining in another way which panel is selected by the mouse, and then taking the picked visual to be the viewbox corresponding to the selected panel.

PR welcome for this, see discussion in #1182. Basically we need a way to select `visual_at` vs `viewbox_geometry` based picking. I suspect in many / most cases people will want the latter, and it should be much faster because it will avoid drawing (with all related Python overhead) entirely.

**Comment by Bram94** [comment]
> Are you unable to run your test code on each of these commits separately? Why not isolate to one (and link to it from this list?

I assume the changes are related to commit 2 of those 4 (ENH: Better mesh options), but my code couldn't run on this one and commit 3 because apparently something was missing that was fixed in commit 4.

> Can you see if this is related to or fixed by #1390? It sounds perhaps related, but if it's not, a separate PR would be good (assuming it works and doesn't break anything of course!).

It seems certainly related to #1390, because it is the kind of improvement that one seems to try to obtain in #1390 (but I didn't read it in detail). For me preventing the emission of the 'transform_changed' event after a pan/zoom event doesn't change anything, except for the speed. During some other types of events it is certainly required, like changing the number of panels.

> PR welcome for this, see discussion in #1182. Basically we need a way to select visual_at vs viewbox_geometry based picking. I suspect in many / most cases people will want the latter, and it should be much faster because it will avoid drawing (with all related Python overhead) entirely.

My method is indeed based on viewbox_geometry, and in this manner the picked visual (viewbox) can be determined in a negligible amount of time. I can create a PR for that (and will likely do so in a few days, although I have little experience yet with Github), but I don't know in which cases the visual_at method is desired.

**Comment by davidh-ssec** [comment]
@Bram94 Is it possible for you to produce a small self-contained example that we (@larsoner, @kmuehlbauer, or me) could run to help us debug this issue further? If not, we'll need you to help us narrow down the exact change (not just the commit, but the actual line(s) of code that changed).

**Comment by kmuehlbauer** [comment]
@davidh-ssec I was in contact with @Bram94 regarding that matter per email because I'm  interested in the whole application. Unfortunately I didn't have time to look into this and will not have in the next two weeks.

Sorry for being no help atm.

**Comment by Bram94** [comment]
@davidh-ssec @kmuehlbauer I will likely also don't have the time to do this in the next few weeks before the Christmas break (I am a student), but will certainly come back to this issue when I have again time.

**Comment by djhoese** [comment]
I was googling for an entirely unrelated issue but this issue was one of the results. We have now released 0.6.x with major updates to how PyQt5 works. I'm going to close this with the assumption that it now works. If that isn't true, then we can reopen this. Let me know.
